src/metadata/evaluator/filter/Filters.js

```diff
@@ -77,9 +78,10 @@ export default class Filters {
      * @param onChange
      * @returns {*}
      */
-    static entityFilter(propertyMetadata, model, keyPrefix, metadataEvaluator, reduxProps, onChange) {
+    static entityFilter(propertyMetadata, model, keyPrefix, metadataEvaluator, reduxProps, onChange, globalScope) {
         if (!propertyMetadata) throw new Error('metadata is required');
         if (!model) throw new Error('model is required');
+        if( !globalScope ) globalScope = model
```

Comment:
For the expression which will be evaluated for some nested field the model passed as first param will be only the related field value of rootModel .

**For example:**

Model
```
         let rootModel = {
                person_address: {
                    active: 'home',
                    home: {
                        city: "Nagpur"
                    },
                    office: {
                        city: "Banglore"
                    }
                }
            }
```

Metadata

```
          let metadata = {
                name: 'person_address',
                type: 'entity',
                fields: [
                    {
                        name: 'active'
                    },
                    {
                        name: 'home',
                        type: 'entity',
                        entityName: 'address',
                        visible: (model, formatter, globalScope)=>{
                            return globalScope.person_address.active === 'home'
                        },
                        fields: [
                            {
                                name: 'city',
                                type: 'string'
                            },
                        ]
                    },
                    {
                        name: 'office',
                        type: 'entity',
                        entityName: 'address',
                        visible: (model, formatter, globalScope)=>{
                            return globalScope.person_address.active === 'office'
                        },
                        fields: [
                            {
                                name: 'city',
                                type: 'string'
                            }
                        ]
                    }
                ]
            };
```
 Example for person_address.home.visible field expression
```
#Expression
             visible: (model, formatter, globalScope)=>{
                   return globalScope.person_address.active === 'home'
              }
#Param Values
            model = home: {
                 city: "Nagpur"
            }

            globalScope =  {
                person_address: {
                    active: 'home',
                    home: {
                        city: "Nagpur"
                    },
                    office: {
                        city: "Banglore"
                    }
                }
            }
```

In above code, when ExpressionEvaluator is called on root of the metadata model refers to rootModel but when ExpressionEvaluator is called for any of nested fields, model passed ExpressionEvaluator will not be a rootModel. Instead it will be a related field value of rootModel. In that case if we want to access rootModel then, this globalScope third param can be used which will refer to the rootModel.